src/test/rustdoc/cross-crate-hidden-assoc-trait-items.rs

```diff
@@ -0,0 +1,13 @@
+// Regression test for issue #95717: Hide cross-crate doc(hidden) associated items in trait impls.
+
+#![crate_name = "dependent"]
+// edition:2021
+// aux-crate:dependency=cross-crate-hidden-assoc-trait-items.rs
+
```

Comment:
I'd like to improve this test, it's not really robust. The thing is, I don't really know where to read up on the syntax of this UI test suite (`@has`, …). I've already searched the rustc-dev-guide to no avail.

Not robust in the sense that if someone changes for example how associated types are rendered like printing its bounds , I think the `@!has` rule will succeed even though it shouldn't. But when I only check if `!has` `id`, leaving out the second test string of each line, then the test passes not only for this PR but also for nightly rust w/o my changes!
But I can't move the test strings like `type Assoc = ()` down one line to a separate `@!has` rule: The tests starts passing no matter what.

Reply:
I would have suggested to look at other `@matches` example to see how they're done but that works too. ;)